packages/monaco-editor-wrapper/src/editorVscodeApi.ts

```diff
@@ -0,0 +1,44 @@
+import { updateUserConfiguration } from 'vscode/service-override/configuration';
+import { registerExtension, IExtensionManifest } from 'vscode/extensions';
+import { MonacoEditorWrapper } from './wrapper.js';
+import 'vscode/default-extensions/theme-defaults';
+import { MonacoEditorBase } from './editor.js';
+
+export type VscodeUserConfiguration = {
+    json?: string;
+}
+
+export type EditorVscodeApiConfig = {
+    extension?: IExtensionManifest | object;
+    extensionFiles?: Map<string, URL>;
```

Comment:
Note to myself: This is too limiting. We should allow to pass json strings directly as well.

Reply:
Done, see next commit